pkg/controller/node/status.go

```diff
@@ -352,12 +361,37 @@ func isNodeDone(node *corev1.Node) bool {
 		return false
 	}
 
+	if layered {
+		// The MachineConfig annotations are loaded on boot-up by the daemon which
+		// isn't currently done for the image annotations, so the comparisons here
+		// are a bit more nuanced.
+		cimage, cok := node.Annotations[daemonconsts.CurrentImageAnnotationKey]
```

Comment:
Can you remind me if any of these can exist but be "empty"? For example, if the pool is opted into layering, and then opted out of layering, do these get cleaned up?

Reply:
There is no way to opt a pool out of  layering yet, right?

Reply:
Not at the moment, no. There is a PR here for that piece: https://github.com/openshift/machine-config-operator/pull/4284

Reply:
When reverted, the `desiredImage` and `currentImage` annotations should be cleared.